src/pages/Books/Books.tsx

```diff
@@ -14,16 +16,17 @@ export const Books: React.FC = () => {
   const [books, setBooks] = useState<BookType[]>([]);
   const [isLoading, setIsLoading] = useState(false);
   const [currentPage, setCurrentPage] = useState(1);
+  const [message, setMessage] = useState<string | undefined>();
```

Comment:
You are NOT checking for undefined/null in any of the logic. I mean you don't have `message === undefined` or similar. Therefore we can simplify this and default it to empty string, which I believe won't break any of the logic.

Update the type so the message is not optional, if the message is empty, just pass an empty string in it. Both approaches have advantages and distavanges, I don't have overly strong opinion about this.

```suggestion
  const [message, setMessage] = useState('');
```